### ISSUE TITLE
[FEATURE_REQUEST] Improved documentation for Debian/Ubuntu

### ISSUE THREAD

**Issue by Aerowinder** [Lissy93/dashy#864]
### Is your feature request related to a problem? If so, please describe.

For people unfamiliar with Node/Yarn, the installation instructions provided from the Build from Source instructions (https://dashy.to/docs/deployment#build-from-source) are inadequate. I realize that this is supposed to be used as a Docker image, but Docker doesn't fit into everyone's environment. For instance, I use Docker at home but not at work.

Would you be willing to expand these instructions, and optionally add some scripting into the repo to make installation from source less of a manual process?

### Describe the solution you'd like

I've created some basic scripts already that do work, but may not be the ideal way to do things. These work on Debian and should work on derivatives:

**Install prereqs, Dashy, and create a new user we will use to run the service:**
`cat > /opt/dashy-install.sh << EOF
#prereq - must use LTS version of Node (v16) as v18 is not supported by one of the dependencies.
apt update && apt install -y curl git
curl -fsSL https://deb.nodesource.com/setup_16.x | bash -
apt-get install -y nodejs

#install
mkdir /opt/dashy
cd /opt
git clone https://github.com/Lissy93/dashy.git
cd /opt/dashy
yarn install
yarn build

#create user
useradd dashy -d /opt/dashy -M -r -s "$(which bash)"
chown -R dashy:dashy /opt/dashy
chmod 771 /opt/dashy
EOF`

**Create start/update scripts. dashy-start.sh is used by the systemctl service to start Dashy, dashy-update must be run manually**
`cat > /opt/dashy-start.sh << EOF
#!/bin/sh
cd /opt/dashy
yarn start
EOF

cat > /opt/dashy-update.sh << EOF
#!/bin/sh
systemctl stop dashy
cd /opt/dashy
git pull origin master
yarn upgrade
yarn build
chown -R dashy:dashy /opt/dashy
chmod 771 /opt/dashy
systemctl start dashy
EOF

chmod +x /opt/dashy-install.sh
chmod +x /opt/dashy-start.sh
chmod +x /opt/dashy-update.sh`

**Create systemd service file so you can control Dashy with systemctl start/stop:**
`cat > /etc/systemd/system/dashy.service << EOF
[Unit]
Description=Dashy Daemon
After=network.target

[Service]
User=dashy
Group=dashy
Type=mixed
ExecStart=/opt/dashy-start.sh
TimeoutStopSec=30
KillMode=mixed
Restart=never

[Install]
WantedBy=multi-user.target
EOF

systemctl daemon-reload
systemctl enable dashy.service
systemctl start dashy.service`

### Priority

Low (Nice-to-have)

### Is this something you would be keen to implement

Maybe

**Comment by averrois** [comment]
Is this still open for improvements, or not !

**Comment by CrazyWolf13** [comment]
> Is this still open for improvements, or not !

It's GitHub, yeah just comment!

Will look into this in the next time, however I will probably wait for 2.1.2 because there are many fixes.

**Comment by averrois** [comment]
@CrazyWolf13 
Yeah, I am kind new to open source lol, okay then I will wait until 2.1.2, then if I find any issues that I could contribute too, it will be an honor to do so

**Comment by TheMrDec** [comment]
Here in the big 26 looking at the docs and trying to set up bare metal. Yarn does not know what to do with this project. Have we removed bare metal support completely or is there a breaking change in yarn that I would need to accommodate?

**Comment by lissy93** [comment]
I can revisit this, I'll spin up a Debian box ands check that the docs are correct (they might not be, since there's been a lot of changes since they were written)

I can update them, but am cautious of distributing scripts for this, as the usage will vary depending on your system configuration, and making them too compressive risks becoming out-dated , and is quite a lot of maintenance overhead.

What are you using (if anything) to manage your debian self-hosted services? If it's a proxmox vm, there's a good community script for this

If you've got specific errors or something not working, let me know and i can help

**Comment by TheMrDec** [comment]
I have an Ubuntu 24.04 LTS VM set up under a proxmox high availability cluster. I went through the process of switching the source for node.js to the node source repository and setting the version number to 20. Following the guide for bare metal I cloned the repository successfully and verified yarn was installeda ia npm but was not able to run yarn build. When running yarn build response that I get from the machine is that there is no file named build. This is why I suspect there might be a change to yarn that requires a different command syntax. It could also be that I just don't know how to use yarn. This is the second or third time that I have ever used yarn and the most recent time was a few years ago, iirc.

As far as scripts go, I'm not interested in a script. I have a tendency to have to edit those to tailor installs to my lab anyway as the lab is a replicant of a production environment and has a lot of odd setup choices spawning from a little bit of tech debt.

You specifically mentioned Debian. Is this something that's only ever been tested on Debian and not Ubuntu as well? If that's the case, I can always just switch to a Debian build.

**Comment by lissy93** [comment]
> When running yarn build response that I get from the machine is that there is no file named build

For the solution to that one, see: #1

You need to install Node.js. Don't run `apt get yarn`, as it installs a totally different random package!

**Comment by TheMrDec** [comment]
I'm guessing that Ubuntu must come with whatever that other yarn package is pre-installed. I did not run apt-get yarn and yet that other yarn package is installed via apt. When running, npm install yarn, it verifies that the yarn dependency is installed, but I'm guessing that running yarn build calls the apt package instead of the npm package. Am not at the lab right now but I should be in about an hour and will see what I can do about calling the correct package.

**Comment by lissy93** [comment]
Done in #2214 

I've re-written all the deployment docs in https://dashy.to/docs/deployment/bare-metal 😊

... but, to be honest after going through what was there before, everything was actually up-to-date and correct still. So if it's still not got all the details you'd like, then you can either submit a PR (to edit [this file](https://github.com/lissy93/dashy/edit/master/docs/deployment/bare-metal.md)) or let me specifically what was missing.

---

i also completed a bunch of the guides for the other deployment methods:
<a href="https://dashy.to/docs/deployment/">
<img width="700" alt="Image" src="https://github.com/user-attachments/assets/1f6c0b7b-29dd-4d28-939f-84da9630c9bd" />
</a>
